src/zmq/zmqpublishnotifier.cpp

```diff
@@ -86,6 +86,15 @@ bool CZMQAbstractPublishNotifier::Initialize(void *pcontext)
             return false;
         }
 
+        const int so_keepalive_option {1};
+        rc = zmq_setsockopt(psocket, ZMQ_TCP_KEEPALIVE, &so_keepalive_option, sizeof(so_keepalive_option));
+        if (rc != 0)
```

Comment:
updated the formatting here to be on the same line. i had to force myself to do it because that's what the dev notes say to do. but it was hard because so much of the zmq code is of the other style :)